src/batch/src/executor/manager.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use async_stream::try_stream;
+use futures::stream::StreamExt;
+use minitrace::prelude::*;
+use risingwave_common::catalog::Schema;
+use risingwave_common::error::RwError;
+use tokio::select;
+use tokio::sync::watch::Receiver;
+
+use crate::executor::{BoxedDataChunkStream, BoxedExecutor, Executor};
+use crate::task::ShutdownMsg;
+
+/// `ManagerExecutor` build on top of the underlying executor. For now, it do two things:
+/// 1. the duration of performance-critical operations will be traced, such as open/next/close.
+/// 2. receive shutdown signal
+pub struct ManagerExecutor {
+    child: BoxedExecutor,
+    /// Description of input executor
+    input_desc: String,
+    shutdown_rx: Receiver<ShutdownMsg>,
+}
+
+impl ManagerExecutor {
+    pub fn new(
+        child: BoxedExecutor,
+        input_desc: String,
+        shutdown_rx: Receiver<ShutdownMsg>,
+    ) -> Self {
+        Self {
+            child,
+            input_desc,
+            shutdown_rx,
+        }
+    }
+}
+
+impl Executor for ManagerExecutor {
+    fn schema(&self) -> &Schema {
+        self.child.schema()
+    }
+
+    fn identity(&self) -> &str {
+        self.child.identity()
+    }
+
+    fn execute(mut self: Box<Self>) -> BoxedDataChunkStream {
+        let return_err = |err: RwError| -> Result<String, RwError> { Err(err) };
+
+        try_stream! {
```

Comment:
There is the bug for `futures-async-stream::try_stream ` so that we can't use select in it. So we have to use `async_stream::try_stream` instead. 

(Is there any better way?